source/common/http/alternate_protocols_cache_impl.cc

```diff
@@ -83,7 +117,7 @@ void AlternateProtocolsCacheImpl::setAlternatives(const Origin& origin,
     protocols_.erase(iter);
   }
   protocols_[origin] = protocols;
-  if (key_value_store_) {
+  if (key_value_store_ && !from_cache) {
```

Comment:
Up to you, but since we're loading from cache instead of calling setAlternativesImpl() we could just do:

protocols_[origin] = protocols

after loading from cache.

Reply:
Yeah I wonder if we could split `setAlternatives` into two functions, one for updating protocols_ and one that updates key_value_store_. That way we could avoid splitting the logic of this function via a bool param? 

Not sure how much of this max_protocols/max_entries handling is required for the cache lookup case